nbsite/pyodide/__init__.py

```diff
@@ -243,69 +284,105 @@ def terminate(cls, *args):
         """
         Terminates a running process.
         """
-        if not cls._current_process:
-            return
-        try:
-            cls._conn.send({'type': 'close'})
-            cls._current_process.join()
-        except Exception:
-            cls._current_process.terminate()
-        finally:
-            cls._current_process = None
+        for state in cls._state.values():
+            if state['process'] is None:
+                return
+            try:
+                state['conn'].send({'type': 'close'})
+                state['process'].join()
+            except Exception:
+                state['process'].terminate()
```

Comment:
It could be nice to output this exception; if something weird happens, it will be hard to debug here.

Reply:
Do we care if or why process termination errors out? Seems inconsequential.

Reply:
The Sphinx process shutdown will terminate the processes anyway.